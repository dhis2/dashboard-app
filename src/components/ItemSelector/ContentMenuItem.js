import React from 'react';
import { MenuItem, colors } from '@dhis2/ui-core';
import i18n from '@dhis2/d2-i18n';

import { getItemIcon } from '../../modules/itemTypes';
import LaunchIcon from '../../icons/Launch';

import classes from './styles/ContentMenuItem.module.css';

const LaunchLink = ({ url }) => (
    <a
        onClick={e => e.stopPropagation()}
        className={classes.launchLink}
        target="_blank"
        rel="noopener noreferrer"
        href={url}
    >
        <LaunchIcon />
    </a>
);

const InsertButton = () => (
    <button className={classes.buttonInsert}>{i18n.t('Insert')}</button>
);

const ContentMenuItem = ({ type, name, onInsert, url }) => {
    const ItemIcon = getItemIcon(type);

    return (
        <MenuItem
            dense
            onClick={onInsert}
            label={
                <div className={classes.menuItem}>
                    <div className={classes.label}>
                        <ItemIcon
                            style={{ margin: '6px', fill: colors.grey600 }}
                        />
                        <span>{name}</span>
                        {url ? <LaunchLink url={url} /> : null}
                    </div>
                    <InsertButton />
                </div>
            }
        />
    );
};

export default ContentMenuItem;
