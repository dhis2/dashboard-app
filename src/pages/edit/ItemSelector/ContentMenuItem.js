import { visTypeIcons } from '@dhis2/analytics'
import i18n from '@dhis2/d2-i18n'
import { MenuItem, colors, IconLaunch16 } from '@dhis2/ui'
import PropTypes from 'prop-types'
import React from 'react'
import { getItemIcon, VISUALIZATION } from '../../../modules/itemTypes.js'
import classes from './styles/ContentMenuItem.module.css'

const LaunchLink = ({ url }) => (
    <a
        onClick={(e) => e.stopPropagation()}
        className={classes.launchLink}
        target="_blank"
        rel="noopener noreferrer"
        href={url}
    >
        <IconLaunch16 color={colors.grey700} />
    </a>
)

LaunchLink.propTypes = {
    url: PropTypes.string,
}

const InsertButton = () => (
    <button className={classes.buttonInsert}>{i18n.t('Insert')}</button>
)

const ContentMenuItem = ({ type, name, onInsert, url, visType }) => {
    const ItemIcon = getItemIcon(type)

    let renderedItemIcon
    if (type === VISUALIZATION) {
        const VisualizationIcon = visTypeIcons[visType]
        renderedItemIcon = <VisualizationIcon color={colors.grey600} />
    } else {
        renderedItemIcon = <ItemIcon color={colors.grey600} />
    }

    return (
        <MenuItem
            onClick={onInsert}
            icon={renderedItemIcon}
            label={
                <div className={classes.menuItem}>
                    <div className={classes.label}>
                        <span>{name}</span>
                        {url ? <LaunchLink url={url} /> : null}
                    </div>
                    <InsertButton />
                </div>
            }
            dataTest={`menu-item-${name}`}
        />
    )
}

ContentMenuItem.propTypes = {
    name: PropTypes.string,
    type: PropTypes.string,
    url: PropTypes.string,
    visType: PropTypes.string,
    onInsert: PropTypes.func,
}

export default ContentMenuItem
