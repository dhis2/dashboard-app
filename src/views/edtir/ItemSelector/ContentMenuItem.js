import React from 'react'
import { MenuItem, colors } from '@dhis2/ui'
import i18n from '@dhis2/d2-i18n'
import PropTypes from 'prop-types'
import { visTypeIcons, VIS_TYPE_SCATTER } from '@dhis2/analytics'

import ScatterIcon from './assets/ScatterIcon'
import LaunchIcon from './assets/Launch'
import { getItemIcon, VISUALIZATION } from '../../../modules/itemTypes'

import classes from './styles/ContentMenuItem.module.css'

const LaunchLink = ({ url }) => (
    <a
        onClick={e => e.stopPropagation()}
        className={classes.launchLink}
        target="_blank"
        rel="noopener noreferrer"
        href={url}
    >
        <LaunchIcon />
    </a>
)

LaunchLink.propTypes = {
    url: PropTypes.string,
}

const InsertButton = () => (
    <button className={classes.buttonInsert}>{i18n.t('Insert')}</button>
)

const ContentMenuItem = ({ type, name, onInsert, url, visType }) => {
    const ItemIcon = getItemIcon(type)

    let renderedItemIcon
    if (type === VISUALIZATION) {
        renderedItemIcon =
            visType === VIS_TYPE_SCATTER ? (
                <ScatterIcon />
            ) : (
                visTypeIcons[visType]
            )
    } else {
        renderedItemIcon = <ItemIcon style={{ fill: colors.grey600 }} />
    }

    return (
        <MenuItem
            dense
            onClick={onInsert}
            label={
                <div className={classes.menuItem}>
                    <div className={classes.label}>
                        <span style={{ marginRight: '6px' }}>
                            {renderedItemIcon}
                        </span>
                        <span>{name}</span>
                        {url ? <LaunchLink url={url} /> : null}
                    </div>
                    <InsertButton />
                </div>
            }
            dataTest={`menu-item-${name}`}
        />
    )
}

ContentMenuItem.propTypes = {
    name: PropTypes.string,
    type: PropTypes.string,
    url: PropTypes.string,
    visType: PropTypes.string,
    onInsert: PropTypes.func,
}

export default ContentMenuItem
