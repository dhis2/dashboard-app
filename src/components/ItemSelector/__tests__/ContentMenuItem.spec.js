import React from 'react';
import { shallow } from 'enzyme';
import ContentMenuItem from '../ContentMenuItem';

describe('ContentMenuItem', () => {
    let props;
    const wrapper = () => shallow(<ContentMenuItem {...props} />);

    beforeEach(() => {
        props = {
            type: 'pony',
            name: 'Pinkie Pie',
            onInsert: jest.fn(),
        };
    });

    it('has onClick action on the Menu Item', () => {
        const menuItem = wrapper();

        expect(menuItem.prop('onClick')).toEqual(props.onInsert);
    });

    it('has a LaunchLink when url is provided', () => {
        props.url = 'http://ponies-r-us.com';

        expect(wrapper()).toMatchSnapshot();
    });

    it('does not have LaunchLink if no url provided', () => {
        expect(wrapper()).toMatchSnapshot();
    });
});
