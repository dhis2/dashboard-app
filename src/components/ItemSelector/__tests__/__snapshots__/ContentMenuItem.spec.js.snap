// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentMenuItem does not have LaunchLink if no url provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentMenuItem
    name="Pinkie Pie"
    onInsert={[MockFunction]}
    type="pony"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "dense": true,
      "label": <div
        className="menuItem"
      >
        <div
          className="label"
        >
          <pure(NotInterestedIcon)
            style={
              Object {
                "fill": "#6e7a8a",
                "margin": "6px",
              }
            }
          />
          <span>
            Pinkie Pie
          </span>
        </div>
        <InsertButton />
      </div>,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dense": true,
        "label": <div
          className="menuItem"
        >
          <div
            className="label"
          >
            <pure(NotInterestedIcon)
              style={
                Object {
                  "fill": "#6e7a8a",
                  "margin": "6px",
                }
              }
            />
            <span>
              Pinkie Pie
            </span>
          </div>
          <InsertButton />
        </div>,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ContentMenuItem has a LaunchLink when url is provided 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContentMenuItem
    name="Pinkie Pie"
    onInsert={[MockFunction]}
    type="pony"
    url="http://ponies-r-us.com"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "dense": true,
      "label": <div
        className="menuItem"
      >
        <div
          className="label"
        >
          <pure(NotInterestedIcon)
            style={
              Object {
                "fill": "#6e7a8a",
                "margin": "6px",
              }
            }
          />
          <span>
            Pinkie Pie
          </span>
          <LaunchLink
            url="http://ponies-r-us.com"
          />
        </div>
        <InsertButton />
      </div>,
      "onClick": [MockFunction],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "dense": true,
        "label": <div
          className="menuItem"
        >
          <div
            className="label"
          >
            <pure(NotInterestedIcon)
              style={
                Object {
                  "fill": "#6e7a8a",
                  "margin": "6px",
                }
              }
            />
            <span>
              Pinkie Pie
            </span>
            <LaunchLink
              url="http://ponies-r-us.com"
            />
          </div>
          <InsertButton />
        </div>,
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
